build_tools/bazel/iree_trace_runner_test.bzl

```
@@ -6,7 +6,222 @@
 
 """Macros for defining tests that use a trace-runner."""
 
-def iree_generated_trace_runner_test(**kwargs):
-    # TODO: implement this. For now, it's only parsed by bazel_to_cmake.py, so
-    # the iree_generated_check_test's are just omitted in bazel builds.
-    pass
+load("//iree/tools:compilation.bzl", "iree_bytecode_module")
+load("//build_tools/bazel:run_binary_test.bzl", "run_binary_test")
+
+def iree_trace_runner_test(
+        name,
+        src,
+        module,
+        target_backend,
+        driver,
+        trace_runner,
+        trace,
+        compiler_flags = [],
+        runner_args = [],
+        opt_tool = "//iree/tools:iree-opt",
+        opt_flags = [],
+        tags = [],
+        timeout = None,
+        **kwargs):
+    """Creates a test running a custom trace-runner on a trace file (yaml).
+
+    Args:
+        name: Name of the target
+        src: mlir source file to be compiled to an IREE module.
+        target_backend: target backend to compile for.
+        driver: driver to run the module with.
+        compiler_flags: additional flags to pass to the compiler. Bytecode
+            translation and backend flags are passed automatically.
+        runner_args: additional args to pass to the trace-runner program. The driver
+            and input file flags are passed automatically.
+        tags: Additional labels to apply to the test. The package path and
+            "driver=${DRIVER}" are added automatically.
```

This isn't true for Bazel. You can use wildcards for Bazel so there's no need to label things by package

```suggestion
        tags: Additional labels to apply to the test. "driver=${DRIVER}" is added 
            automatically.
```

---

Thanks!